### DISCUSSION TITLE
Niches for imaging based spatial transcriptomics data

### DISCUSSION THREAD

**Discussion by chhatbarchintan** [msraredon/NICHES#14] [Q&A]
@msraredon ...hi,
I am using CosMx platform from nanostring for spatial transcriptomics analysis. Its different than spot based (10x) spatial transcriptomics in that wach molecule has x and y co-ordinates and cells are ploygons with centroids. Cell can and will have different radius and area. I used niches to analyse a seurat object which i have created from the spatial data that i have and when  `RunNICHES` on the seurat object i get and error like this:

Set cell types as Identity of object internally

 Subsetting to populations with greater than 0 cells

 25960 distinct cells from 14 celltypes to be analyzed

 Limiting ground truth to genes within dataset

 Mapping against 583 ground truth signaling mechanisms
Error: Cannot find 'x' in this Seurat object

i believe the error is due to different data structure between a seura tobject created from visium data and the CosMx data. do you have any idea how to solve this?
Thanks

**Comment by msraredon** [comment]
Interesting. I don't think @jcyang34 or I have encountered this question before. Sounds like we should address it.

Would you be able to send me a small piece of data with reproducible code, and I can see what I can do on my end?

S

**Reply by chhatbarchintan** [reply]
@msraredon...it was a simple solution....just extarcted the co-ordinates and added to metadata using this. Thanks for your quick response though.

spatial.locs = as.data.frame(nano.obj@images[["fov"]]@boundaries[["centroids"]]@coords)
nano.obj$x <- spatial.locs$x
nano.obj$y <- spatial.locs$y
